docker/Dockerfile.context

```diff
@@ -0,0 +1,49 @@
+# This dockerfile creates a thin docker image that can be used for making custom
+# cross Dockerfiles that live in other git repositories.
+# It has been uploaded to `alsuren/cross-context` so it can be used like so:
+
+## Fetch this image and use it as `context`:
+# FROM alsuren/cross-context:latest as context
```

Comment:
- [ ] What's the process for getting this uploaded somewhere more official?

Reply:
Our CI has access to dockerhub, see

https://github.com/rust-embedded/cross/blob/2af06e2a871f4cf467eb241a40d6ebef0cc43a8f/azure-pipelines.yml#L123

For example

Reply:
Hrm. The more I think about it, maybe this Dockerfile.context thing should be punted to another PR.

- It isn't strictly necessary for my use-case (cross-compiling something that links against dbus). You can get away with something like:
```
FROM rustembedded/cross:armv7-unknown-linux-gnueabihf-0.2.1 as context

RUN dpkg --add-architecture armhf && \
    apt-get update && \
    apt-get install -y libssl-dev:armhf libdbus-1-dev:armhf

ENV PKG_CONFIG_PATH=/usr/lib/arm-linux-gnueabihf/pkgconfig
```
- adding Dockerfile.context means that all of your build-scripts are suddenly exposed like public API, and are easy to break accidentally.
- Testing the context image is fiddly and intrusive, and requires more thought.